lib/grandpa/message.go

```diff
@@ -105,31 +72,17 @@ type VoteMessage struct {
 	Message *SignedMessage
 }
 
-// Decode SCALE decodes the data into a VoteMessage
-func (v *VoteMessage) Decode(r io.Reader) (err error) {
-	v.Round, err = common.ReadUint64(r)
-	if err != nil {
-		return err
-	}
-
-	v.SetID, err = common.ReadUint64(r)
-	if err != nil {
-		return err
-	}
-
-	v.Message = new(SignedMessage)
-	err = v.Message.Decode(r)
-	return err
-}
-
 // Type returns voteType
 func (v *VoteMessage) Type() byte {
 	return voteType
 }
 
+// Index Returns VDT index
+func (v *VoteMessage) Index() uint { return 0 }
```

Comment:
```suggestion
func (vm *VoteMessage) Index() uint { return 0 }
```
Update all the other receiver functions too.